server/src/main/java/com/broadcom/lsp/cobol/core/messages/LocaleStoreImpl.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.messages;
+
+import com.broadcom.lsp.cobol.service.delegates.communications.Communications;
+import com.google.common.collect.Iterables;
+import com.google.gson.JsonPrimitive;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.lsp4j.MessageType;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+
+/**
+ * Concrete implementation for {@link LocaleStore}
+ *
+ * <p>Stores the locale for entire application and is single point of contact for all locale related
+ * issues. This class update application locale, notify subscriber in case of locale update and
+ * returns currently set locale.
+ */
+@Slf4j
+@Singleton
+public final class LocaleStoreImpl implements LocaleStore {
+  private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+  private final Lock readLock = readWriteLock.readLock();
+  private final Lock writeLock = readWriteLock.writeLock();
+  private final Locale defaultLocale = new Locale("en", "en");
+  private Communications communication;
+  private LocaleEnum supportedLocale;
+  private List<Consumer<Locale>> notifyList = new ArrayList<>();
+
+  @Inject
+  public LocaleStoreImpl(Communications communications) {
+    this.communication = communications;
+  }
+
+  /**
+   * Return the application current locale.
+   *
+   * @return {@link LocaleEnum}
+   */
+  @Override
+  public LocaleEnum getSupportedLocale() {
+    readLock.lock();
+    try {
+      if (Objects.isNull(supportedLocale))
+        return Enum.valueOf(LocaleEnum.class, defaultLocale.getCountry().toUpperCase());
+      return supportedLocale;
```

Comment:
This "return" should be moved to end of method. Regarding to finally would be more readable

Reply:
moving return to the last stmt in the method. return in finally might give a sonar issue of - "Jump statements should not occur in "finally" blocks"

Reply:
Of course I wrote, end-of-method.